usl_models/usl_models/flood_ml/dataset.py

```diff
@@ -94,7 +172,8 @@ def generator():
             ),
         ),
     )
-    dataset = dataset.batch(batch_size)
+    if batch_size:
```

Comment:
What does this branch protect against? The batch_size being 0?

Reply:
Yes, when batch size is zero it avoids adding an extra batch dimension, which was what caused the batch prediction error.

Reply:
Okay! A comment to that effect would be helpful.

Reply:
a test for that case would also be helpful so we don't hit this error again!

Reply:
Separated this into another PR.